### PR TITLE
Add vesting contracts as templates. Contracts credit to Alex

### PR DESCRIPTION
**Pull request by OptimusOpus**
Contracts credit to Alex aka xiangzhe0812

### CODE REVIEW

File: contracts/SaleVesting.sol [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    // Time at which vesting begins.
+    uint256 claimStartTime;
+}
+
+// Client's requirement not yet finalized, still waiting.
+contract SaleVesting is Ownable {
+    using SafeERC20 for ERC20;
+
+    address public immutable self;
+    address public immutable RITE;
```
Comment (bitcoinbrisbane) [inline contracts/SaleVesting.sol:33]: Shouldnt be RITE

File: contracts/SaleVesting.sol [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    uint256 public TGEDate;
+
+    constructor(address _RITE, uint256 _TGEDate) {
+        self = address(this);
+        RITE = _RITE;
```
Comment (bitcoinbrisbane) [inline contracts/SaleVesting.sol:38]: And here.

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Needs to have unit tests and more generic names.

**Comment by rellfy** [comment]
SaleVesting and TeamVesting are pretty much the same contract. For the template we can add just one not to be redundant

**Pull request closed without merging**